core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/internal/DefaultSyncSignRequest.java

```diff
@@ -44,14 +57,21 @@ public static final class BuilderImpl<IdentityT extends Identity>
         implements SyncSignRequest.Builder<IdentityT> {
 
         // Used to enable consumer builder pattern in HttpSigner.sign()
-        public BuilderImpl() {
+        private BuilderImpl() {
         }
 
         // Used by SyncSignRequest#builder() where identity is passed as parameter, to avoid having to pass Class<IdentityT>.
-        public BuilderImpl(IdentityT identity) {
+        private BuilderImpl(IdentityT identity) {
             super(identity);
         }
 
+        private BuilderImpl(DefaultSyncSignRequest<IdentityT> request) {
+            properties(request.properties)
+                .identity(request.identity)
+                .payload(request.payload)
+                .request(request.request);
```

Comment:
nit: the indentation from the 2nd line makes this odd to read. if that's the default style because of chaining the methods, maybe don't chain them as that works too here.